src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerChangeService.java

```diff
@@ -0,0 +1,291 @@
+
+        if (publicPrivateMoPair.getRight() != null) {
+            managingOfficerChange.setAppointmentId(publicPrivateMoPair.getRight().getManagingOfficerAppointmentId());
+        }
+
+        if (publicPrivateMoPair.getLeft() != null && publicPrivateMoPair.getLeft().getAppointedOn() == null) {
+            officer.setStartDate(managingOfficerCorporateDto.getStartDate());
+        }
+
+        boolean hasChange = setCommonAttributes(changeManager,
+                managingOfficerCorporateDto.getServiceAddress(), managingOfficerCorporateDto.getResignedOn());
+
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                managingOfficerCorporateDto.getName(),
+                CompanyOfficerApi::getName,
+                CorporateManagingOfficer::setName
+        );
+
+        hasChange |= changeManager.compareAndBuildLeftChange(
+                managingOfficerCorporateDto.getPrincipalAddress(),
+                CompanyOfficerApi::getPrincipalOfficeAddress,
+                TypeConverter::addressDtoToAddress,
+                ComparisonHelper::equals,
+                CorporateManagingOfficer::setRegisteredOffice
+        );
+
+        hasChange |= changeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getContactFullName(),
+                ManagingOfficerDataApi::getContactNameFull,
+                CorporateManagingOfficer::setContactName
+        );
+
+        hasChange |= changeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getContactEmail(),
+                ManagingOfficerDataApi::getContactEmailAddress,
+                CorporateManagingOfficer::setEmail
+        );
+
+        officer.setCompanyIdentification(new CompanyIdentification());
+
+        IdentificationApi publicIdentification = publicPrivateMoPair.getLeft() == null ? null
+                : publicPrivateMoPair.getLeft().getIdentification();
+
+        ChangeManager<CompanyIdentification, CompanyOfficerApi, IdentificationApi> companyIdentificationChangeManager
+                = new ChangeManager<>(officer.getCompanyIdentification(),
+                new ImmutablePair<>(publicPrivateMoPair.getLeft(), publicIdentification));
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getLegalForm(),
+                IdentificationApi::getLegalForm,
+                CompanyIdentification::setLegalForm);
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getLawGoverned(),
+                IdentificationApi::getLegalAuthority,
+                CompanyIdentification::setGoverningLaw);
+
+        hasChange |= companyIdentificationChangeManager.compareAndBuildRightChange(
+                managingOfficerCorporateDto.getPublicRegisterName(),
+                IdentificationApi::getPlaceRegistered,
```

Comment:
Please see latest main branch. I believe this should be passed as a change to placeRegistered. It's a bit strange but the "place" is the register name (while registerLocation seems to be used somewhat inconsistently). For now, please align with BO changes on main.